lxd/auth/driver_rbac.go

```diff
@@ -338,54 +371,113 @@ func (r *rbac) DeleteProject(projectID int64) error {
 }
 
 // RenameProject renames an existing project resource in RBAC.
-func (r *rbac) RenameProject(projectID int64, name string) error {
-	return r.AddProject(projectID, name)
+func (r *rbac) RenameProject(ctx context.Context, projectID int64, oldName string, newName string) error {
+	return r.AddProject(ctx, projectID, newName)
 }
 
-// UserAccess returns a UserAccess struct for the user.
-func (r *rbac) UserAccess(username string) (*UserAccess, error) {
-	r.permissionsLock.Lock()
-	defer r.permissionsLock.Unlock()
+// StopService stops the periodic status checker.
+func (r *rbac) StopService(ctx context.Context) error {
+	r.ctxCancel()
+	return nil
+}
 
-	// Check whether the permissions are cached.
-	_, cached := r.permissions[username]
+type rbacResource struct {
+	Identifier string `json:"identifier"`
+	Name       string `json:"name"`
+}
 
-	if !cached {
-		_ = r.syncPermissions(username)
-	}
+type rbacResourcePost struct {
+	LastSyncID *string        `json:"last-sync-id"`
+	Updates    []rbacResource `json:"updates,omitempty"`
+	Removals   []string       `json:"removals,omitempty"`
+}
 
-	// Checked if the user exists.
-	permissions, ok := r.permissions[username]
-	if !ok {
-		return nil, errUnknownUser
-	}
+type rbacResourcePostResponse struct {
+	SyncID string `json:"sync-id"`
+}
+
+type rbacStatus struct {
+	LastChange string `json:"last-change"`
+}
 
-	// Prepare the response.
-	access := UserAccess{
-		Admin:    shared.ValueInSlice("admin", permissions[""]),
-		Projects: map[string][]string{},
+// startStatusCheck runs a status checking loop.
+func (r *rbac) startStatusCheck() {
+	var status rbacStatus
+
+	// Figure out the new URL.
+	u, err := url.Parse(r.apiURL)
+	if err != nil {
+		logger.Errorf("Failed to parse RBAC url: %v", err)
+		return
 	}
 
-	for k, v := range permissions {
-		// Skip the global permissions.
-		if k == "" {
-			continue
-		}
+	u.Path = path.Join(u.Path, "/api/service/v1/changes")
 
-		// Look for project name.
-		for projectName, resourceID := range r.resources {
-			if k != resourceID {
+	go func() {
+		for {
+			if r.ctx.Err() != nil {
+				return
+			}
+
+			if status.LastChange != "" {
+				values := url.Values{}
+				values.Set("last-change", status.LastChange)
+				u.RawQuery = values.Encode()
+			}
+
+			req, err := http.NewRequestWithContext(r.ctx, "GET", u.String(), nil)
+			if err != nil {
+				if err == context.Canceled {
+					return
+				}
+
+				logger.Errorf("Failed to prepare RBAC query: %v", err)
+				return
+			}
+
+			resp, err := r.client.Do(req)
+			if err != nil {
+				if err == context.Canceled {
+					return
+				}
+
+				// Handle server/load-balancer timeouts, errors aren't properly wrapped so checking string.
+				if strings.HasSuffix(err.Error(), "EOF") {
+					continue
+				}
+
+				logger.Errorf("Failed to connect to RBAC, re-trying: %v", err)
```

Comment:
Something like  "Failed RBAC status check, failed connecting to RBAC service, retrying: %v"